e2e/tests/smoke/smoke_test_helper.go

```diff
@@ -0,0 +1,416 @@
+package smoke
+
+import (
+	"errors"
+	"github.com/rook/rook/e2e/rook-test-framework/clients"
+	"github.com/rook/rook/e2e/rook-test-framework/contracts"
+	"github.com/rook/rook/e2e/rook-test-framework/enums"
+	"github.com/rook/rook/e2e/rook-test-framework/utils"
+)
+
+type SmokeTestHelper struct {
+	platform   enums.RookPlatformType
+	rookclient *clients.RookClient
+	rookHelp   *utils.RookHelper
+	k8sHelp    *utils.K8sHelper
+}
+
+type blockTestData struct {
+	name       string
+	size       int
+	mountpath  string
+	pvcDefPath string
+	podDefPath string
+}
+
+type fileTestData struct {
+	name       string
+	mountpath  string
+	podname    string
+	podDefPath string
+}
+
+type objetUserData struct {
+	userId      string
+	displayname string
+	emailId     string
+}
+
+type objctConnectionData struct {
+	aws_endpoint          string
+	aws_host              string
+	aws_secret_key_id     string
+	aws_secret_access_key string
+}
+
+func createObjectUserData(userid string, displayname string, emailid string) objetUserData {
+	return objetUserData{userid, displayname, emailid}
+}
+
+func createObjectConnectionData(endpoint string, host string, secretid string, secretkey string) objctConnectionData {
+	return objctConnectionData{endpoint, host, secretid, secretkey}
+}
+
+func CreateSmokeTestClient(platform enums.RookPlatformType) (*SmokeTestHelper, error) {
+
+	rc, err := getRookClient(platform)
+
+	return &SmokeTestHelper{platform: platform,
+		rookclient: rc,
+		rookHelp:   utils.CreateRookHelper(),
+		k8sHelp:    utils.CreatK8sHelper()}, err
+
+}
+
+func getRookClient(platform enums.RookPlatformType) (*clients.RookClient, error) {
+	return clients.CreateRook_Client(platform)
+
+}
+func (integrationTestClient *SmokeTestHelper) GetBlockClient() contracts.Irook_block {
+	return integrationTestClient.rookclient.Get_Block_client()
+}
+
+func (integrationTestClient *SmokeTestHelper) GetFileSystemClient() contracts.Irook_filesystem {
+	return integrationTestClient.rookclient.Get_FileSystem_client()
+}
+
+func (integrationTestClient *SmokeTestHelper) GetObjectClient() contracts.Irook_object {
+	return integrationTestClient.rookclient.Get_Object_client()
+}
+
+func (integrationClient *SmokeTestHelper) getBlockTestData() (blockTestData, error) {
+	switch integrationClient.platform {
+	case enums.Kubernetes:
+		return blockTestData{"block-test",
+			1048576,
+			"/tmp/rook1",
+			"../../../test-data/smoke-test-data/smoke_pool_sc_pvc.tmpl",
```

Comment:
instead of using these relative paths, could the test be given a parameter which is the test-data folder? then the test only needs a much less brittle path